### ISSUE TITLE
"running bonnet_model_compiler.par on "20180408-102900.pb" (ubuntu18.04 facenet)

### ISSUE THREAD

**Issue by liyoung1992** [google/aiyprojects-raspbian#411]
Hi!
I'm try to deployed facenet  model in VisionKit!and I'm in trouble!
I run the compiler using:
 ./bonnet_model_compiler.par --frozen_graph_path="./20180408-102900.pb" --input_tensor_name="input" --output_tensor_names="embeddings" --input_tensor_size=160 --output_graph_path="./facenet.binarypro" --debug

but it always returns an "unexpected error":
2018-06-15 10:56:46.200831: I external/org_tensorflow/tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before Removing unused ops: 3578 operators, 5806 arrays (0 quantized)
2018-06-15 10:56:46.401776: I external/org_tensorflow/tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 3578 operators, 5806 arrays (0 quantized)
2018-06-15 10:56:46.403070: F external/org_tensorflow/tensorflow/contrib/lite/toco/graph_transformations/resolve_batch_normalization.cc:42] Check failed: IsConstantParameterArray(*model, bn_op->inputs[1]) && IsConstantParameterArray(*model, bn_op->inputs[2]) && IsConstantParameterArray(*model, bn_op->inputs[3]) Batch normalization resolution requires that mean, multiplier and offset arrays be constant.

facenet:
[facenet](https://github.com/davidsandberg/facenet)
.pb:
https://drive.google.com/open?id=1R77HmFADxe87GmoLwzfgMu_HY0IhcyBz

Thanks!

**Comment by weiran-work** [comment]
@liyoung1992 
Their version of FaceNet is based on Inception+Resnet, which is too big and resource demanding to run on VisionKit. If the FaceNet is trained with MobileNet v1/v2 with supported input size and depthwise multiplier (for example, 160x160 dw=0.5 or 192x192 dw=1.0 and etc), you should be able to deploy it on VisionKit.

**Comment by liyoung1992** [comment]
@weiranzhao thanks！
Since I'm just an ordinary soft developer, I don't know too much about AI. This may be beyond my ability. Will aiy provide some trained models, or provide a detailed method of how to compile the model?In this case, estimates use more people！
thanks a lot！

**Comment by weiranzhao** [comment]
@liyoung1992 
We recently added some instructions / short tutorial on how to retrain customized model that can be deployed on vision kit. Check this page out.
https://aiyprojects.withgoogle.com/vision/#makers-guide--tensorflow-model-compiler
EXTENSION TO POET TUTORIAL is highly recommended.

I'm closing this issue for now. 
If you ran into issues, feel free to open new issues.

**Comment by weiran-work** [comment]
@liyoung1992
We recently added some instructions / short tutorial on how to retrain customized model that can be deployed on vision kit. Check this page out.
https://aiyprojects.withgoogle.com/vision/#makers-guide--tensorflow-model-compiler
EXTENSION TO POET TUTORIAL is highly recommended.

If you ran into issues, feel free to open new issues.

**Comment by liyoung1992** [comment]
thanks！

**Comment by JakeMalis** [comment]
@weiranzhao 
Hi Weiran,

I am just starting out on Tensorflow & AIY Vision Kit. I am able to completely build Tensorflow for Poets on Colab as well as on a GCE instance (NVIDIA K80 GPU). I do this with 160 size and 0.5.

I then compile the PB file into a BinaryProto and upload it (and the retrained_labels.txt) to the RPI. When I run the mobilenet_based_classifier.py file, the program crashes after 5 seconds of seeing what the camera sees and I get the following 3 errors:
PI Camera Runtime: No Data received from sensors
AIY Vision Inference: Camera is not running
AIY Drivers Spiccom internal error

I also found a thread with others having the issue from 3 days ago: https://github.com/google/aiyprojects-raspbian/issues/432

Any ideas?

Thanks.
